File: backend/internal/models/device.go
package models

import "time"

type DeviceStatus string

const (
	DeviceStatusOnline  DeviceStatus = "ONLINE"
	DeviceStatusOffline DeviceStatus = "OFFLINE"
)

type Device struct {
	ID            int64        `json:"id"`
	DeviceID      string       `json:"device_id" gorm:"uniqueIndex;not null" validate:"required"`
	DeviceName    string       `json:"device_name" validate:"required"`
	SerialNumber  string       `json:"serial_number" validate:"required"`
	OSVersion     *string      `json:"os_version,omitempty"`
	IPAddress     *string      `json:"ip_address,omitempty"`
	Location      *string      `json:"location,omitempty"`
	Status        DeviceStatus `json:"status"`
	LastSeen      *time.Time   `json:"last_seen,omitempty"`
	LastOnlineAt  *time.Time   `json:"last_online_at,omitempty"`
	LastOfflineAt *time.Time   `json:"last_offline_at,omitempty"`
	CreatedAt     time.Time    `json:"created_at"`
	UpdatedAt     time.Time    `json:"updated_at"`
}

File: frontend/src/types/device.ts
export type DeviceStatus = "ONLINE" | "OFFLINE";

export interface Device {
    id: number;
    device_id: string;
    device_name: string;
    serial_number: string;
    os_version?: string | null;
    ip_address?: string | null;
    location?: string | null;
    status: DeviceStatus;
    last_seen?: string | null;
    last_online_at?: string | null;
    last_offline_at?: string | null;
    created_at: string;
    updated_at: string;
}

export interface DeviceStatusChangedEvent {
    type: "DEVICE_STATUS_CHANGED";
    device_id: string;
    status: DeviceStatus;
    device: Device;
}